src/main/java/net/minecraftforge/event/enchanting/EnchantmentLevelSetEvent.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.event.enchanting;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.common.eventhandler.Event;
+
+/**
+ * Fired when the enchantment level is set for a slot in the enchanting table. The {@link #level} is set to the vanilla value 
+ * and can be modified by this event handler.
+ *
+ * The {@link #slot} is used to determine which slot's enchantment level is being set. The {@link power} is a number from 0-15
+ * and indicates how many bookshelves surround the enchanting table. The {@link itemStack} representing the item being enchanted
+ * is also available.
+ */
+public class EnchantmentLevelSetEvent extends Event
+{
+    private final World world;
+    private final BlockPos pos;
+    private final int enchantNum;
```

Comment:
I was using the term slot before, but it was being confused as having something to do with inventory slots. I agree though, that enchantNum still isn't perfect. What about enchantRow? or enchantLine?